test/unittest/conftest.py

```diff
@@ -1,113 +1,52 @@
+from superduperdb.base.build import build_datalayer
 from superduperdb.base.datalayer import Datalayer
 from superduperdb.base.document import Document
 from superduperdb.components.dataset import Dataset
 from superduperdb.components.listener import Listener
-from superduperdb.components.metric import Metric
 from superduperdb.components.vector_index import VectorIndex
-from superduperdb.ext.numpy.encoder import array
 from superduperdb.ext.pillow.encoder import pil_image
 
 n_data_points = 250
 
+LOCAL_TEST_N_DATA_POINTS = 5
 
-@pytest.fixture()
-def empty(test_db: Datalayer):
-    yield test_db
-
-
-@pytest.fixture()
-def metric(empty):
-    empty.add(Metric(identifier='p@1', object=PatK(1)))
-    yield
-    empty.remove('metric', 'p@1', force=True)
+MONGOMOCK_URI = 'mongomock:///test_db'
 
 
 @pytest.fixture
-def random_data_factory(float_tensors_32):
-    float_tensor = float_tensors_32.encoders['torch.float32[32]']
-
-    def _factory(number_data_points=n_data_points):
-        data = []
-        for i in range(number_data_points):
-            x = torch.randn(32)
-            y = int(random.random() > 0.5)
-            z = torch.randn(32)
-            data.append(
-                Document(
-                    {
-                        'x': float_tensor(x),
-                        'y': y,
-                        'z': float_tensor(z),
-                    }
-                )
-            )
-        float_tensors_32.execute(
-            Collection('documents').insert_many(data),
-            refresh=False,
-        )
-        return float_tensors_32
-
-    return _factory
-
-
-@pytest.fixture()
-def random_data(random_data_factory):
-    return random_data_factory()
-
-
-@pytest.fixture()
-def random_arrays(arrays):
-    float_array = arrays.encoders['numpy.float32[32]']
-    data = []
-    for i in range(n_data_points):
-        x = numpy.random.randn(32).astype(numpy.float32)
-        y = int(random.random() > 0.5)
-        data.append(Document({'x': float_array(x), 'y': y}))
-
-    arrays.execute(Collection('documents').insert_many(data, refresh=False))
-    yield arrays
-    arrays.execute(Collection('documents').delete_many({}))
-
-
-@pytest.fixture()
-def a_single_insert(float_tensors_32):
-    float_tensor = float_tensors_32.encoders['torch.float32[32]']
-    x = torch.randn(32)
-    y = int(random.random() > 0.5)
-    z = torch.randn(32)
-    return Document(
-        {
-            'x': float_tensor(x),
-            'y': y,
-            'z': float_tensor(z),
-        }
+def valid_dataset():
+    d = Dataset(
+        identifier='my_valid',
+        select=Collection('documents').find({'_fold': 'valid'}),
+        sample_size=100,
     )
+    return d
 
 
-@pytest.fixture()
-def an_insert(float_tensors_32):
-    float_tensor = float_tensors_32.encoders['torch.float32[32]']
+def add_random_data(
+    data_layer: Datalayer,
+    collection_name: str = 'documents',
+    number_data_points: int = n_data_points,
```

Comment:
isnt this global var>

Reply:
No, just a helper func for adding random data

Reply:
`add_random_data` don't use `pytest.fixture` decorator